compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangAnonymousModelHelper.java

```diff
@@ -97,6 +101,34 @@ public String getNextAnonymousTypeKey(PackageID packageID) {
         return ANON_TYPE + UNDERSCORE + nextValue;
     }
 
+    public String getNextAnonymousTypeKey(PackageID packageID, Stack<String> suffixes) {
+        if (suffixes.isEmpty()) {
+            return getNextAnonymousTypeKey(packageID);
+        }
+        return createAnonTypeName(suffixes, packageID);
+    }
+
+    private String createAnonTypeName(Stack<String> suffixes, PackageID pkgId) {
+        String name = ANON_TYPE;
+        for (int i = suffixes.size() - 1; i >= 0; i--) {
+            name = name.concat(suffixes.elementAt(i) + DOLLAR);
+        }
+        if (!anonTypesNamesPerPkg.containsKey(pkgId)) {
```

Comment:
I believe we can replace the rest of this function like below. Check the difference between id and name when id == 0 case if we need it.

```Java
Map<String, Integer> anonTypesNames = anonTypesNamesPerPkg.putIfAbsent(pkgId, new HashMap<>());
Integer id = anonTypesNames.putIfAbsent(name, 0);
anonTypesNames.put(name, id + 1);
return name + UNDERSCORE + id;
```

Reply:
@KRVPerera we need to have a null check here right?

Reply:
I think we need something like the below here.
```Java
anonTypesNamesPerPkg.putIfAbsent(pkgId, new HashMap<>());
Map<String, Integer> anonTypesNames = anonTypesNamesPerPkg.get(pkgId);
anonTypesNames.putIfAbsent(name, 0);
Integer id = anonTypesNames.get(name);
anonTypesNames.put(name, id + 1);
return name + UNDERSCORE + id;
```

Reply:
May be `computeIfAbsent` is better

Reply:
```
Map<String, Integer> anonTypesNames = anonTypesNamesPerPkg.computeIfAbsent(pkgId, pkgId -> new HashMap<>());
anonTypesNames.putIfAbsent(name, 0);
Integer id = anonTypesNames.get(name);
anonTypesNames.put(name, id + 1);
return name + UNDERSCORE + id;
```